<launch>
	<!-- RGB-D Camera Driver -->
	<include file="$(find velodyne_pointcloud)/launch/VLP16_points.launch"/>

	<!-- Global Params -->
	<group ns="perception">
		<param name="RGB" value="false"/>
	</group>

	<!-- Statistical Outlier Remover Node -->
	<node name="StatRemover" pkg="perception" type="stat_remove" output="screen">
		<param name="mean_k" value="50"/>
		<param name="std_dev_multiplier_thresh" value="1"/>
	</node>
	
	<!-- Cropper Node -->
	<node name="Cropper" pkg="perception" type="crop" output="screen">
		<!-- Crop Params -->
		<param name="crop_min_x" value="-2.0"/>
		<param name="crop_min_y" value="-2.0"/>
		<param name="crop_min_z" value="-2.0"/>
		<param name="crop_max_x" value="5.0"/>
		<param name="crop_max_y" value="5.0"/>
		<param name="crop_max_z" value="5.0"/>
	</node>

	<!-- Downsampler Node -->
	<node name="Downsampler" pkg="perception" type="downsample" output="screen">
		<param name="voxel_size" value="0.01"/>
	</node>

	<!-- Segmentation Node -->
	<node name="Segmenter" pkg="perception" type="segment" output="screen">
		<!-- Planar Segment Params -->
		<param name="distance_threshold" value="0.01"/>
		<param name="axis" value="Z"/>
		<param name="epsilon_angle" value="0.0"/>
		<param name="distance_above_plane" value="0.005"/>

		<!-- Euclidean Cluster Params -->
		<param name="cluster_tolerance" value="0.06"/>
		<param name="min_cluster_size" value="200"/>
		<param name="max_cluster_size" value="10000"/>
	</node>

	<!-- Perception Visualization Node-->
	<node name="rviz" pkg="rviz" type="rviz" output="screen" args="-d $(find perception)/rviz/example.rviz">
	</node>

	<!-- Dynamic Reconfigure Node -->
	<node name="Reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" output="screen"/> 
	
	<!-- Publish TF frames -->
	<node name="map_to_camera" pkg="tf" type="static_transform_publisher" args="0 0 0 -0.5 -0.5 -0.5 0.5 map camera_link 100">
	</node>

	<node name="map_to_VLP" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 1 map velodyne 100">
	</node> 

	<!--node name="map_to_camera" pkg="tf" type="static_transform_publisher" args="0 0 0 0.707 0 0 -0.707 map camera_color_optical_frame 100">
	</node-->

	<!--node name="map_to_camera" pkg="tf" type="static_transform_publisher" args="0 0 0
		0 0 0 1 map camera_color_optical_frame 100">
	</node-->
</launch>